File: div11/01.js
console.time("time")

let lst = []

i = 102345678
while (lst.length === 0) {
    if ((new Set(String(i).split(""))).size === 9) {
        if (i % 11 === 0) {
            lst.push(i)
            break    
        }
    }
    i += 1
}

while (i <= 987654321) {
    i += 11
    if ((new Set(String(i).split(""))).size == 9) 
        lst.push(i)
}


console.log("len(lst) =", lst.length)
console.log(lst[0], lst[lst.length-1])

console.timeEnd("time")

File: 00-Основы научных знаний/div11/01.py
import time

start = time.monotonic()

lst = []

i = 102345678
while len(lst) == 0:
    if len(set(str(i))) == len(str(i)):
        if i % 11 == 0:
            lst.append(i)
            break
    i += 1

while i <= 987654321:
    i += 11
    if len(set(str(i))) == len(str(i)):
        lst.append(i)

print("len(lst) =", len(lst))
print(lst[0], lst[-1])

finish = time.monotonic()
result = finish - start
print(round(result,2), "сек")
